src/main/java/com/github/leroyguillaume/keycloak/bcrypt/BCryptPasswordHashProvider.java

```diff
@@ -48,7 +55,13 @@ public void close() {
     @Override
     public boolean verify(final String rawPassword, final PasswordCredentialModel credential) {
         final String hash = credential.getPasswordSecretData().getValue();
-        final BCrypt.Result verifier = BCrypt.verifyer(BCrypt.Version.VERSION_2A).verify(rawPassword.toCharArray(), hash.toCharArray());
-        return verifier.verified;
+        char[] pw = rawPassword.toCharArray();
+        if (pw.length > BCrypt.Version.MAX_PW_LENGTH_BYTE - 1) {
```

Comment:
In the case of the `verify` I think we should take a dual-path verification approach. Instead of relying on the input to determine the bcrypt version I would recommend we:

1. Run a verify with the most common algorithm first (BCrypt.Version.VERSION_2Y_NO_NULL_TERMINATOR).
2. If (1) fails we fallback to the current algorithm (BCrypt.Version.VERSION_2A).

I think this will give us piece of mind that no matter how the password was hashed (with the termination char or not) we have support for it.